### ISSUE TITLE
The given key was not present in the dictionary

### ISSUE THREAD

**Issue by jch0182** [ONLYOFFICE/Docker-CommunityServer#145]
After getting everything up and running (docs, community server, and control panel), I am getting a couple of odd errors when attempting to access documents or calendar. 

Calendar shows a pop-up dialog when accessing the calendar and with each new calendar or event that is created:
The given key '28271' was not present in the dictionary.

Documents shows a modal overlay in the upper right corner that says:
The given key '25698' was not present in the dictionary.

Documents will not allow for any new documents to be created, uploaded, etc. So completely unusable.

Some other errors I'm seeing as well:

svcNotify.log:

```
2022-03-29 21:16:55,215 ERROR [16] ASC - System.Collections.Generic.KeyNotFoundException: The given key '31084' was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x0001e] in <f2b3ab7dfff746f594d2ef5b16ec3c90>:0 
  at MySql.Data.MySqlClient.MySqlField.SetFieldEncoding () [0x00034] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlField.set_CharacterSetIndex (System.Int32 value) [0x00007] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.NativeDriver.GetColumnData (MySql.Data.MySqlClient.MySqlField field) [0x000ad] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.NativeDriver.GetColumnsData (MySql.Data.MySqlClient.MySqlField[] columns) [0x00004] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.Driver.GetColumns (System.Int32 count) [0x0001c] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.ResultSet.LoadColumns (System.Int32 numCols) [0x00000] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.ResultSet..ctor (MySql.Data.MySqlClient.Driver d, System.Int32 statementId, System.Int32 numCols) [0x0002a] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.Driver.NextResult (System.Int32 statementId, System.Boolean force) [0x00039] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlDataReader.NextResult () [0x000d5] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlDataReader.Close () [0x00039] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlCommand.ResetReader () [0x00020] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader (System.Data.CommandBehavior behavior) [0x0032c] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader () [0x00000] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery () [0x00032] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlTransaction.Rollback () [0x00051] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlConnection.CloseFully () [0x0002f] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlConnection.Close () [0x00045] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlConnection.Dispose (System.Boolean disposing) [0x00009] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at System.ComponentModel.Component.Dispose () [0x00000] in <88f564ea69dd4dc8ba9bf979e48d5996>:0 
  at (wrapper remoting-invoke-with-check) System.ComponentModel.Component.Dispose()
  at ASC.Common.Data.DbManager.Dispose () [0x00041] in <96a81f2b5ce744ef90c32874e89bb46b>:0 
  at ASC.Notify.DbWorker.GetMessages (System.Int32 count) [0x001c2] in <29d0ec7edc9c488b8bd6b226190bf73b>:0 
  at ASC.Notify.NotifySender.ThreadManagerWork () [0x0004f] in <29d0ec7edc9c488b8bd6b226190bf73b>:0  System.Collections.Generic.KeyNotFoundException: The given key '31084' was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x0001e] in <f2b3ab7dfff746f594d2ef5b16ec3c90>:0 
  at MySql.Data.MySqlClient.MySqlField.SetFieldEncoding () [0x00034] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlField.set_CharacterSetIndex (System.Int32 value) [0x00007] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.NativeDriver.GetColumnData (MySql.Data.MySqlClient.MySqlField field) [0x000ad] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.NativeDriver.GetColumnsData (MySql.Data.MySqlClient.MySqlField[] columns) [0x00004] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.Driver.GetColumns (System.Int32 count) [0x0001c] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.ResultSet.LoadColumns (System.Int32 numCols) [0x00000] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.ResultSet..ctor (MySql.Data.MySqlClient.Driver d, System.Int32 statementId, System.Int32 numCols) [0x0002a] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.Driver.NextResult (System.Int32 statementId, System.Boolean force) [0x00039] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlDataReader.NextResult () [0x000d5] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlDataReader.Close () [0x00039] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlCommand.ResetReader () [0x00020] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader (System.Data.CommandBehavior behavior) [0x0032c] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader () [0x00000] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery () [0x00032] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlTransaction.Rollback () [0x00051] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlConnection.CloseFully () [0x0002f] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlConnection.Close () [0x00045] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlConnection.Dispose (System.Boolean disposing) [0x00009] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at System.ComponentModel.Component.Dispose () [0x00000] in <88f564ea69dd4dc8ba9bf979e48d5996>:0 
  at (wrapper remoting-invoke-with-check) System.ComponentModel.Component.Dispose()
  at ASC.Common.Data.DbManager.Dispose () [0x00041] in <96a81f2b5ce744ef90c32874e89bb46b>:0 
  at ASC.Notify.DbWorker.GetMessages (System.Int32 count) [0x001c2] in <29d0ec7edc9c488b8bd6b226190bf73b>:0 
  at ASC.Notify.NotifySender.ThreadManagerWork () [0x0004f] in <29d0ec7edc9c488b8bd6b226190bf73b>:0 

```

web.files.log:
```

2022-03-29 22:07:24,989 ERROR [125] ASC.Files - System.Collections.Generic.KeyNotFoundException: The given key '12548' was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x0001e] in <f2b3ab7dfff746f594d2ef5b16ec3c90>:0 
  at MySql.Data.MySqlClient.MySqlField.SetFieldEncoding () [0x00034] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlField.set_CharacterSetIndex (System.Int32 value) [0x00007] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.NativeDriver.GetColumnData (MySql.Data.MySqlClient.MySqlField field) [0x000ad] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.NativeDriver.GetColumnsData (MySql.Data.MySqlClient.MySqlField[] columns) [0x00004] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.Driver.GetColumns (System.Int32 count) [0x0001c] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.ResultSet.LoadColumns (System.Int32 numCols) [0x00000] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.ResultSet..ctor (MySql.Data.MySqlClient.Driver d, System.Int32 statementId, System.Int32 numCols) [0x0002a] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.Driver.NextResult (System.Int32 statementId, System.Boolean force) [0x00039] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlDataReader.NextResult () [0x000d5] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlDataReader.Close () [0x00039] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlDataReader.Dispose (System.Boolean disposing) [0x00003] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlDataReader.Dispose () [0x00000] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery () [0x0004b] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at ASC.Common.Data.AdoProxy.DbCommandProxy.ExecuteNonQuery () [0x00012] in <96a81f2b5ce744ef90c32874e89bb46b>:0 
  at ASC.Common.Data.DataExtensions.ExecuteNonQuery (System.Data.Common.DbCommand command, System.String sql, System.Object[] parameters) [0x00009] in <96a81f2b5ce744ef90c32874e89bb46b>:0 
  at ASC.Common.Data.DbManager.ExecuteNonQuery (System.String sql, System.Object[] parameters) [0x00006] in <96a81f2b5ce744ef90c32874e89bb46b>:0 
  at ASC.Common.Data.DbManagerProxy.ExecuteNonQuery (System.String sql, System.Object[] parameters) [0x00006] in <96a81f2b5ce744ef90c32874e89bb46b>:0 
  at ASC.Files.Core.Data.TagDao.GetNewTags (System.Guid subject, System.Action insert) [0x000d6] in <f783d77b75674b6a980575e87f6cfb3a>:0 
  at ASC.Files.Core.Data.TagDao.GetNewTags (System.Guid subject, System.Collections.Generic.IEnumerable`1[T] fileEntries) [0x00014] in <f783d77b75674b6a980575e87f6cfb3a>:0 
  at ASC.Files.Thirdparty.ProviderDao.ProviderTagDao.GetNewTags (System.Guid subject, System.Collections.Generic.IEnumerable`1[T] fileEntries) [0x00007] in <ed797b27a1904408b7a52fd8727113e9>:0 
  at ASC.Web.Files.Utils.EntryManager.SetFileStatus (System.Collections.Generic.IEnumerable`1[T] files) [0x0003d] in <f783d77b75674b6a980575e87f6cfb3a>:0 
  at ASC.Web.Files.Utils.EntryManager.GetEntries (ASC.Files.Core.IFolderDao folderDao, ASC.Files.Core.IFileDao fileDao, ASC.Files.Core.Folder parent, System.Int32 from, System.Int32 count, ASC.Files.Core.FilterType filter, System.Boolean subjectGroup, System.Guid subjectId, System.String searchText, System.Boolean searchInContent, System.Boolean withSubfolders, ASC.Files.Core.OrderBy orderBy, System.Int32& total) [0x00977] in <f783d77b75674b6a980575e87f6cfb3a>:0 
  at ASC.Web.Files.Services.WCFService.FileStorageServiceController.GetFolderItems (System.String parentId, System.Int32 from, System.Int32 count, ASC.Files.Core.FilterType filter, System.Boolean subjectGroup, System.String ssubject, System.String searchText, System.Boolean searchInContent, System.Boolean withSubfolders, ASC.Files.Core.OrderBy orderBy) [0x000fc] in <f783d77b75674b6a980575e87f6cfb3a>:0  System.Collections.Generic.KeyNotFoundException: The given key '12548' was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x0001e] in <f2b3ab7dfff746f594d2ef5b16ec3c90>:0 
  at MySql.Data.MySqlClient.MySqlField.SetFieldEncoding () [0x00034] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlField.set_CharacterSetIndex (System.Int32 value) [0x00007] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.NativeDriver.GetColumnData (MySql.Data.MySqlClient.MySqlField field) [0x000ad] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.NativeDriver.GetColumnsData (MySql.Data.MySqlClient.MySqlField[] columns) [0x00004] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.Driver.GetColumns (System.Int32 count) [0x0001c] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.ResultSet.LoadColumns (System.Int32 numCols) [0x00000] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.ResultSet..ctor (MySql.Data.MySqlClient.Driver d, System.Int32 statementId, System.Int32 numCols) [0x0002a] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.Driver.NextResult (System.Int32 statementId, System.Boolean force) [0x00039] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlDataReader.NextResult () [0x000d5] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlDataReader.Close () [0x00039] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlDataReader.Dispose (System.Boolean disposing) [0x00003] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlDataReader.Dispose () [0x00000] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery () [0x0004b] in <8235ed0ad41343b3a11f3dc973b415b6>:0 
  at ASC.Common.Data.AdoProxy.DbCommandProxy.ExecuteNonQuery () [0x00012] in <96a81f2b5ce744ef90c32874e89bb46b>:0 
  at ASC.Common.Data.DataExtensions.ExecuteNonQuery (System.Data.Common.DbCommand command, System.String sql, System.Object[] parameters) [0x00009] in <96a81f2b5ce744ef90c32874e89bb46b>:0 
  at ASC.Common.Data.DbManager.ExecuteNonQuery (System.String sql, System.Object[] parameters) [0x00006] in <96a81f2b5ce744ef90c32874e89bb46b>:0 
  at ASC.Common.Data.DbManagerProxy.ExecuteNonQuery (System.String sql, System.Object[] parameters) [0x00006] in <96a81f2b5ce744ef90c32874e89bb46b>:0 
  at ASC.Files.Core.Data.TagDao.GetNewTags (System.Guid subject, System.Action insert) [0x000d6] in <f783d77b75674b6a980575e87f6cfb3a>:0 
  at ASC.Files.Core.Data.TagDao.GetNewTags (System.Guid subject, System.Collections.Generic.IEnumerable`1[T] fileEntries) [0x00014] in <f783d77b75674b6a980575e87f6cfb3a>:0 
  at ASC.Files.Thirdparty.ProviderDao.ProviderTagDao.GetNewTags (System.Guid subject, System.Collections.Generic.IEnumerable`1[T] fileEntries) [0x00007] in <ed797b27a1904408b7a52fd8727113e9>:0 
  at ASC.Web.Files.Utils.EntryManager.SetFileStatus (System.Collections.Generic.IEnumerable`1[T] files) [0x0003d] in <f783d77b75674b6a980575e87f6cfb3a>:0 
  at ASC.Web.Files.Utils.EntryManager.GetEntries (ASC.Files.Core.IFolderDao folderDao, ASC.Files.Core.IFileDao fileDao, ASC.Files.Core.Folder parent, System.Int32 from, System.Int32 count, ASC.Files.Core.FilterType filter, System.Boolean subjectGroup, System.Guid subjectId, System.String searchText, System.Boolean searchInContent, System.Boolean withSubfolders, ASC.Files.Core.OrderBy orderBy, System.Int32& total) [0x00977] in <f783d77b75674b6a980575e87f6cfb3a>:0 
  at ASC.Web.Files.Services.WCFService.FileStorageServiceController.GetFolderItems (System.String parentId, System.Int32 from, System.Int32 count, ASC.Files.Core.FilterType filter, System.Boolean subjectGroup, System.String ssubject, System.String searchText, System.Boolean searchInContent, System.Boolean withSubfolders, ASC.Files.Core.OrderBy orderBy) [0x000fc] in <f783d77b75674b6a980575e87f6cfb3a>:0 
```

**Comment by Carazyda** [comment]
Hello @jch0182 Hello @jch0182 Please describe how did you install Onlyoffice Workspace and what is server OS?

**Comment by jch0182** [comment]
I should have added all of that as well, I apologize. This is running inside Docker on an Ubuntu 20.04.4 LTS OS. I used the install instructions from https://helpcenter.onlyoffice.com/installation/workspace-install-docker-integrated.aspx but converted to a docker-compose file instead - with one exception, that I already have a database up and running, so put in the credentials for that database instead.

Which ended up being the problem. I had MariaDB up and running, which no matter what I did, would not install correctly. I stood up MySQL instead per the instructions and everything is working. I would really rather just have a single database, but nothing I did would make the install scripts work with MariaDB, so I made the switch.

**Comment by Carazyda** [comment]
Sorry for the late reply. It looks like you are using Mariadb 10.6 and in this version they have changed to work from [utf-8](https://jira.mariadb.org/browse/MDEV-8334). And so we have problems with stable work. With mariadb version 10.5 everything works successfully.

**Comment by HoffmannTom** [comment]
Hello,
I have the same problem with MySQL 8.
The tables and database are utf8mb3. OS is Ubuntu 20.04 LTS.
We are using the docker images with an external MySQL 8 database. It stopped working two weeks ago.
Can we add some connection parameters for fixing the charset issue?
Any other options available?
Thanks in advance!

**Comment by Carazyda** [comment]
Hello @HoffmannTom The problem is in the new version of MySQL Server 8.0.29. We will fix this in a new version that will be released soon.

**Comment by HoffmannTom** [comment]
Yes... thats excactly the version we are using: Ver 8.0.29-0ubuntu0.20.04.3 for Linux on x86_64
Thanks for the information :)

**Comment by HoffmannTom** [comment]
As a temporary workaround, it worked for me to copy .Net 4.5.2 MySQL drivers into the docker-instance:

docker cp /tmp/MySql.Data.dll 9978c1880d45://var/www/onlyoffice/Services/MailWatchdog/MySql.Data.dll
docker cp /tmp/MySql.Data.dll 9978c1880d45:/var/www/onlyoffice/Services/Jabber/MySql.Data.dll
docker cp /tmp/MySql.Data.dll 9978c1880d45:/var/www/onlyoffice/Services/MailCleaner/MySql.Data.dll
docker cp /tmp/MySql.Data.dll 9978c1880d45:/var/www/onlyoffice/Services/TeamLabSvc/MySql.Data.dll
docker cp /tmp/MySql.Data.dll 9978c1880d45:/var/www/onlyoffice/Services/MailAggregator/MySql.Data.dll
docker cp /tmp/MySql.Data.dll 9978c1880d45:/var/www/onlyoffice/WebStudio/bin/MySql.Data.dll
docker cp /tmp/MySql.Data.dll 9978c1880d45:/var/www/onlyoffice/ApiSystem/bin/MySql.Data.dll

(Adjust docker instance ID)

Mono-drivers are available at https://dev.mysql.com/downloads/connector/net/8.0.html 
Just in case that someone needs a quick fix.

Greetings, Thomas

**Comment by Carazyda** [comment]
The issue has been resolved in version 12.1.
